src/drivers/intel/cavs/mn.c

```diff
@@ -55,7 +55,7 @@ int mn_set_mclk(uint16_t mclk_id, uint32_t mclk_rate)
 	int clk_index = -1;
 	int ret = 0;
 
-	if (mclk_id > 1) {
+	if (mclk_id >= DAI_NUM_SSP_MCLK) {
```

Comment:
off-by-one?

the mclk ID is 0 or 1, here you test >= 2

Reply:
Because it's test for *invalid* id, 0 1 are valid with 2 you'll get error etc

Reply:
need more coffee